graphql-server/src/schemas/custom.ts

```diff
@@ -758,3 +801,150 @@ async function publishNatsMessage(topic: string, message: string) {
     );
   }
 }
+
+function buildPatientsQueryBody({
+  searchVals,
+  filter,
+}: {
+  searchVals: QueryDashboardPatientsArgs["searchVals"];
+  filter: QueryDashboardPatientsArgs["filter"];
+}) {
+  const fieldsToSearch = [
+    "smilePatientId",
+    "cmoPatientId",
+    "dmpPatientId",
+    "cmoSampleIds",
+    "consentPartA",
+    "consentPartC",
+  ];
+
+  const searchFilters = searchVals?.length
+    ? "WHERE " +
+      fieldsToSearch
+        .map((field) => `${field} =~ '(?i).*(${searchVals.join("|")}).*'`)
+        .join(" OR ")
+    : "";
+
+  const patientsQueryBody = `
+    MATCH (p:Patient)
+
+    // Get patient aliases
+    OPTIONAL MATCH (p)<-[:IS_ALIAS]-(cmoPa:PatientAlias {namespace: 'cmoId'})
+    OPTIONAL MATCH (p)<-[:IS_ALIAS]-(dmpPa:PatientAlias {namespace: 'dmpId'})
+
+    // Get the latest SampleMetadata of each Sample
+    OPTIONAL MATCH (p)-[:HAS_SAMPLE]->(s:Sample)-[:HAS_METADATA]->(sm:SampleMetadata)
+    WITH
+      p,
+      cmoPa,
+      dmpPa,
+      s,
+      collect(sm) AS allSampleMetadata,
+      max(sm.importDate) AS latestImportDate
+    WITH
+      p,
+      cmoPa,
+      dmpPa,
+      s,
+      [sm IN allSampleMetadata WHERE sm.importDate = latestImportDate][0] AS latestSm
+
+    // Get the CMO Sample IDs and additionalProperties JSONs from SampleMetadata
+    WITH
+      p,
+      cmoPa,
+      dmpPa,
+      s,
+      CASE
+        WHEN latestSm.cmoSampleName IS NOT NULL THEN latestSm.cmoSampleName
+            ELSE latestSm.primaryId
+        END AS cmoSampleId,
+      apoc.convert.fromJsonMap(latestSm.additionalProperties) AS latestSmAddlPropsJson
+
+    // Implode/aggregate the different arrays
+    WITH
+      p,
+      cmoPa,
+      dmpPa,
+      collect(s) as samples,
+      collect(cmoSampleId) AS cmoSampleIds,
+      collect(latestSmAddlPropsJson.\`consent-parta\`) as consentPartAs,
+      collect(latestSmAddlPropsJson.\`consent-partc\`) as consentPartCs
```

Comment:
I assume this will return null if these fields don't exist in the additional properties? Can you confirm?

Reply:
Good question. Confirmed. Accessing non-existent/made-up properties in a node returns `null` ([source](https://neo4j.com/developer/kb/understanding-non-existent-properties-and-null-values/)).

If the only input to `collect( )` is `null`, as in `collect(null)`, then it returns an empty array `[ ]`.

Going some lines below this section, we have `consentPartAs[0]`. If `consentPartAs` is an empty array `[ ]`, then accessing an index on it also returns a `null`.